pkg/osutils/filesystem/darwin.go

```diff
@@ -0,0 +1,73 @@
+//go:build darwin
+
+/*
+ * Copyright 2022 Aspect Build Systems, Inc.
+ *
+ * Licensed under the aspect.build Commercial License (the "License");
+ * you may not use this file except in compliance with the License.
+ * Full License text is in the LICENSE file included in the root of this repository.
+ */
+
+package filesystem
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+)
+
+const ChmodPath = "/bin/chmod"
```

Comment:
I'm thinking that we want to use `chmod` instead of the absolute path. If you call just `chmod`, then we take advantage of how it resolves the `PATH` environment variable. I can see how in this case it's rarely the case that the user would want to override it via `PATH` but there's no reason why we would make such a choice of enforcing `/bin/chmod`. This is a comment about a generalized approach to this kind of situation.